src/main/java/com/sivalabs/ft/features/domain/ProductService.java

```diff
@@ -37,7 +39,7 @@ public Long createProduct(CreateProductCommand cmd) {
 
     @Transactional
     public void updateProduct(UpdateProductCommand cmd) {
-        var product = productRepository.findByCode(cmd.code()).orElseThrow();
+        var product = productRepository.findByCode(cmd.code()).orElseThrow(() -> new EntityNotFoundException("Product %s not found".formatted(cmd)));
```

Comment:
Would it be better to create our own `ResourceNotFoundException` so that JPA exceptions don't leak into web layer?

Reply:
Yep, let's do it.